controllers/kustomization_controller.go

```diff
@@ -88,6 +88,7 @@ type KustomizationReconciler struct {
 	ControllerName        string
 	statusManager         string
 	NoCrossNamespaceRefs  bool
+	NoRemoteBases         bool
```

Comment:
```suggestion
	AllowRemoteBases         bool
```
I think the kind of negative logic here might confuse users so a positive logic is more user-friendly from my point of view.

Reply:
> I think the kind of negative logic here might confuse users so a positive logic is more user-friendly from my point of view.

IMO the flags prefixed with `--no` tell people that a major feature that they rely on is going to be disabled. Having `allow` for something that was always available in Flux it's way more confusing.

Reply:
I was actually thinking that a default of `false` for `AllowRemoteBases` was what we wanted in the service of security by default.

Reply:
> I was actually thinking that a default of false for AllowRemoteBases was what we wanted in the service of security by default.

Are you seriously considering breaking 99% of Flux users clusters?

Reply:
Also commented to this effect below. There may be a compromise between breaking users now (before GA) vs getting setting insecure defaults that will be around for the long-term. I suggested a potential approach for the security focus folks.

Reply:
This is not the right forum to propose something as impactful as this. Given that we didn't disabled cross-refs by default, doing so for remote bases would require a RFC where we gather feedback from the Flux community. The current implementation and naming convention follows the cross-refs one.

Reply:
> > I was actually thinking that a default of false for AllowRemoteBases was what we wanted in the service of security by default.
> 
> Are you seriously considering breaking 99% of Flux users clusters?

😆 No, that was not my intention. I kind of thought about the flag in isolation and didn't think about the upgrade flow potentially breaking things w/o proper supervision (i.e. setting the flag accordingly to retain existing behaviour).

In that case it looks like we don't have much of a choice now.